### ISSUE TITLE
NVARCHAR(MAX) on Sql2000

### ISSUE THREAD

**Issue by simoneb** [schambers/fluentmigrator#571]
If I understand correctly Sql2000 does not support `VARCHAR(MAX)` and `.AsString(int.MaxValue)` generates that on Sql2000, according to [this](https://github.com/schambers/fluentmigrator/blob/master/src/FluentMigrator.Runner/Generators/SqlServer/SqlServer2000TypeMap.cs#L44).

Shouldn't it generate `NTEXT` instead?

**Comment by tommarien** [comment]
Good point ! same goes for nvarchar (ntext) and varchar should map to (text)

**Comment by tommarien** [comment]
@simoneb I finally got some time to work on this, but i have a doubt. For instance

.AsString(int.maxvalue) should ,according to [this](https://technet.microsoft.com/en-us/library/aa258271%28v=sql.80%29.aspx), throw a notsupportedexception 

> Variable-length Unicode data with a maximum length of 2^30 - 1 (1,073,741,823) characters.

What do we do here ?
1. keep the happy convention of int.Maxvalue generates max allowed value 
2. have a correct type map

**Comment by ALMMa** [comment]
What about an enum 'StringLength.MaxAvailableLength'?

This way it's possible to deprecate Int32.MaxValue with least impact (a simple search-replace should do the trick).

Filosophically, this makes more sense, as it makes clear that FM is trying to set the field to the max allowed length possible, and not to VARCHAR(2147483647).

**Comment by tommarien** [comment]
@ALMMa I like the idea, but the issue is not with deprecation int.MaxValue

Sql2000 has max NTEXT size up to 1,073,741,823 characters so we have two options here either 

- Allow AsString(int.maxvalue) to map to NTEXT which is handy but theoretically incorrect
- NotSupportedException when we use NTEXT(int.maxvalue)

I vote for the allow with a comment in the code (only for string)

**Comment by ALMMa** [comment]
@tommarien I got that.

Problem is that I got your comments a bit diferent, as if you were asking about .AsString(int.MaxValue) per si and it's mapping regardless of text.

Anyway, some folks might be expecting .AsString to create a varchar field, not a text. They are close but not the same. 

According to [this](http://www.sqlservercentral.com/Forums/Topic1362876-1291-1.aspx) many string functions won't work on text. Although can't guarantee that's 100% accurate, it does make sense and could break existing views/sp/triggers.

For me this should not be changed to text/ntext but instead to varchar(8000) and nvarchar(4000). 
Be advised, however, that I don't use SQL2000.
